src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
+export const bookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      const idx = state.indexOf(action.payload);
+      if (idx !== -1) state.splice(idx, 1);
+    },
+  },
```

Comment:
- [ ] [Optional] This can work fine if the payload is a simple value that is directly stored in the state array. However, if the payload is an object, indexOf may not work as expected, as it performs a strict comparison (===) to find the index. To handle removing an object from the state array correctly. you can consider this code below:
` removeBook: (state, action) => {`
      `const idx = state.findIndex((book) => book.id === action.payload.id);`
   `   if (idx !== -1) state.splice(idx, 1);`
    `},`

Reply:
Thank you